src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<L, R> {
+    private L first;
+    private R second;
 
+    private Pair(L first, R second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public L getFirst() {
+        return first;
+    }
+
+    public R getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
+        return compare(first, pair.first) && compare(second, pair.second);
```

Comment:
try to think of a way to make it a bit more simple

Reply:
fix